src/main/java/de/allcom/services/auth/AuthentificationService.java

```diff
@@ -0,0 +1,155 @@
+public class AuthentificationService {
+
+    private final UserRepository userRepository;
+
+    private final AddressRepository addressRepository;
+
+    private final TokenRepository tokenRepository;
+
+    private final PasswordEncoder passwordEncoder;
+
+    private final JwtService jwtService;
+
+    private final AuthenticationManager authenticationManager;
+
+    public AuthentificationResponse register(UserAddressRegistrationDto request) {
+        if (userRepository.findByEmail(request.getEmail()).isPresent()) {
+            throw new RestException(HttpStatus.CONFLICT,
+                    "User with email " + request.getEmail() + " already exists!");
+        }
+        LocalDateTime now = LocalDateTime.now();
+        var user = User.builder()
+                .firstName(request.getFirstName())
+                .lastName(request.getLastName())
+                .email(request.getEmail())
+                .phoneNumber(request.getPhoneNumber())
+                .companyName(request.getCompanyName())
+                .position(request.getPosition())
+                .hashPassword(passwordEncoder.encode(request.getPassword()))
+                .taxNumber(request.getTaxNumber())
+                .role(Role.CLIENT)
+                .createAt(now)
+                .updateAt(now)
+                .build();
+
+        var savedUser = userRepository.save(user);
+
+        var address = Address.builder()
+                .postIndex(request.getPostIndex())
+                .city(request.getCity())
+                .street(request.getStreet())
+                .houseNumber(request.getHouseNumber())
+                .user(savedUser)
+                .build();
+
+        addressRepository.save(address);
+
+        var jwtToken = jwtService.generateToken(user);
+        savedUserToken(savedUser, jwtToken);
+        return AuthentificationResponse.builder()
+                .id(savedUser.getId())
+                .firstName(savedUser.getFirstName())
+                .lastName(savedUser.getLastName())
+                .email(savedUser.getEmail())
+                .token(jwtToken)
+                .build();
+    }
+
+    public AuthentificationResponse login(AuthentificationRequest request) {
+        if (request.getEmail() == null || request.getPassword() == null) {
+            throw new RestException(HttpStatus.BAD_REQUEST,
+                    "Email and password are required!");
+        } else if (userRepository.findByEmail(request.getEmail()).isEmpty()) {
+            throw new RestException(HttpStatus.NOT_FOUND,
+                    "User with email " + request.getEmail() + " not found!");
+        }
+
+        try {
+            authenticationManager.authenticate(
+                    new UsernamePasswordAuthenticationToken(
+                            request.getEmail(),
+                            request.getPassword()));
+        } catch (Exception e) {
+            throw new RestException(HttpStatus.UNAUTHORIZED, e.getMessage());
```

Comment:
Can you repeat this Exception in postman?

Reply:
Removed this exception. Error interception is implemented in the class SecurityConfiguration